drivers/gpu/drm/bridge/cdns-mhdp.h

```diff
@@ -226,13 +227,26 @@ struct cdns_mhdp_device {
 	struct drm_dp_link link;
 	struct drm_dp_aux aux;
 
+	/*
+	 * Big lock to protect HW access and device data structures.
+	 * With this lock all transactions are atomic and there is no
+	 * interleaving between them. Only IRQ registers are accessed
+	 * without the lock in the IRQ handler. That access is
+	 * exclusive to IRQ handler after the HW is enabled. Before
+	 * that the IRQs are disabled.
+	 */
+	struct mutex mutex;
```

Comment:
It's not very clear what this protects. Everything under cnds_mhdp_device, and all the registers (except irq registers)?
Why it's not enough to protect hw_enabled and bridge_attached?